src/Builders/ComponentParameterBuilder.cs

```diff
@@ -0,0 +1,351 @@
+        /// <returns>A <see cref="ComponentParameterBuilder{TComponent}"/> which can be chained</returns>
+        public ComponentParameterBuilder<TComponent> Add<TValue>(Expression<Func<TComponent, RenderFragment<TValue>?>> parameterSelector, Func<TValue, string> markupFactory)
+        {
+            if (parameterSelector is null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+
+            if (markupFactory is null)
+            {
+                throw new ArgumentNullException(nameof(markupFactory));
+            }
+
+            return Add(parameterSelector, value => renderTreeBuilder => renderTreeBuilder.AddMarkupContent(0, markupFactory(value)));
+        }
+
+        /// <summary>
+        /// Add a strongly typed <see cref="EventCallback" /> parameter with a callback for a component under test.
+        /// </summary>
+        /// <param name="parameterSelector">The parameter selector which defines the parameter to add</param>
+        /// <param name="callback">A callback that returns a <see cref="Task"/>.</param>
+        /// <returns>A <see cref="ComponentParameterBuilder{TComponent}"/> which can be chained</returns>
+        public ComponentParameterBuilder<TComponent> Add(Expression<Func<TComponent, EC>> parameterSelector, Func<Task> callback)
+        {
+            if (parameterSelector is null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+
+            if (callback is null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            return Add(parameterSelector, EC.Factory.Create(this, callback));
+        }
+
+        /// <summary>
+        /// Add a strongly typed <see cref="EventCallback{TValue}" /> parameter with a callback for a component under test.
+        /// </summary>
+        /// <param name="parameterSelector">The parameter selector which defines the parameter to add</param>
+        /// <param name="callback">A callback that takes a <typeparamref name="TValue"/> and returns a <see cref="Task"/>.</param>
+        /// <returns>A <see cref="ComponentParameterBuilder{TComponent}"/> which can be chained</returns>
+        public ComponentParameterBuilder<TComponent> Add<TValue>(Expression<Func<TComponent, EventCallback<TValue>>> parameterSelector, Func<TValue, Task> callback)
+        {
+            if (parameterSelector is null)
+            {
+                throw new ArgumentNullException(nameof(parameterSelector));
+            }
+
+            if (callback is null)
+            {
+                throw new ArgumentNullException(nameof(callback));
+            }
+
+            var (name, isCascading) = GetDetailsFromExpression(parameterSelector);
+            return AddParameterToList(name, EC.Factory.Create(this, callback), isCascading);
+        }

```

Comment:
```suggestion
        public ComponentParameterBuilder<TComponent> Add<TValue>(Expression<Func<TComponent, EventCallback<TValue>>> parameterSelector, Func<TValue, Task> callback) => Add(parameterSelector, EC.Factory.Create(this, callback));
```
I _think_ this should be OK.